From patchwork Wed Jul 20 02:18:02 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Liam Howlett <Liam.Howlett@Oracle.com>
X-Patchwork-Id: 12923275
Return-Path: <owner-linux-mm@kvack.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from kanga.kvack.org (kanga.kvack.org [205.233.56.17])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 6EFA3CCA47F
	for <linux-mm@archiver.kernel.org>; Wed, 20 Jul 2022 02:19:26 +0000 (UTC)
Received: by kanga.kvack.org (Postfix)
	id 863E46B007E; Tue, 19 Jul 2022 22:19:22 -0400 (EDT)
Received: by kanga.kvack.org (Postfix, from userid 40)
	id 79ED96B0080; Tue, 19 Jul 2022 22:19:22 -0400 (EDT)
X-Delivered-To: int-list-linux-mm@kvack.org
Received: by kanga.kvack.org (Postfix, from userid 63042)
	id 616F58E0001; Tue, 19 Jul 2022 22:19:22 -0400 (EDT)
X-Delivered-To: linux-mm@kvack.org
Received: from relay.hostedemail.com (smtprelay0011.hostedemail.com
 [216.40.44.11])
	by kanga.kvack.org (Postfix) with ESMTP id 477336B007E
	for <linux-mm@kvack.org>; Tue, 19 Jul 2022 22:19:22 -0400 (EDT)
Received: from smtpin29.hostedemail.com (a10.router.float.18 [10.200.18.1])
	by unirelay08.hostedemail.com (Postfix) with ESMTP id 28DD31403BE
	for <linux-mm@kvack.org>; Wed, 20 Jul 2022 02:19:22 +0000 (UTC)
X-FDA: 79705871364.29.1B5D8E9
Received: from mx0a-00069f02.pphosted.com (mx0a-00069f02.pphosted.com
 [205.220.165.32])
	by imf16.hostedemail.com (Postfix) with ESMTP id 9B4AF18000A
	for <linux-mm@kvack.org>; Wed, 20 Jul 2022 02:19:21 +0000 (UTC)
Received: from pps.filterd (m0246617.ppops.net [127.0.0.1])
	by mx0b-00069f02.pphosted.com (8.17.1.5/8.17.1.5) with ESMTP id
 26K0JPuv031784;
	Wed, 20 Jul 2022 02:19:19 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : references : in-reply-to : content-type :
 content-transfer-encoding : mime-version; s=corp-2022-7-12;
 bh=46DvMK5GEP4d9GsmJC/9dg2qzjXnaZqUwxzUp3QEP0Q=;
 b=cueXUC4EQzK7TgrKTjSKyPI68veTYdTroViPDma6ykXTK3QqxOxOiGhrljU3zbHlELgr
 DabwHp95OQS2gLGJH3juhf1X7jeX6YYgKAEkbgqIIDFknt2J1JrkWfGyUv71nE4JCD+2
 WmHrgLpsl7KoLH/fbDk/sqCmdRBSDSIHAnRPsdG+Yx7GLOh89IiSZhixJACOIcsPDaJE
 UiMvcJ93Zs0gGXjQlJU7F3Yz/kb3RP09ES8SHYN8bhsC0CorG5bJJclCJvqD/dUdXFx7
 0HepKckqBskgXnojdzaEtVpV3Y459EHdnN+K+pYD5VG2aiXWudmouul2qUzwF0QreCRK HA==
Received: from iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com
 (iadpaimrmta03.appoci.oracle.com [130.35.103.27])
	by mx0b-00069f02.pphosted.com (PPS) with ESMTPS id 3hbnvtg6cd-1
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Wed, 20 Jul 2022 02:19:18 +0000
Received: from pps.filterd
 (iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com [127.0.0.1])
	by iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com (8.17.1.5/8.17.1.5)
 with ESMTP id 26JNeDsO022213;
	Wed, 20 Jul 2022 02:19:17 GMT
Received: from nam11-dm6-obe.outbound.protection.outlook.com
 (mail-dm6nam11lp2170.outbound.protection.outlook.com [104.47.57.170])
	by iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com (PPS) with ESMTPS id
 3hc1hseb6k-2
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Wed, 20 Jul 2022 02:19:17 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=FmVs5iIsVouHZBr6g8TTjXPX4LoNi/lSfvTOoWJ7woLbMr291OGjSufjknXa7wk/dRihtzYThYeUP9aWx0DwlSqxOLMkUzP+Y7/yLZKrSWne0ZAtFB08fv3XZH+3krujEmzrDPKH4ncYqku3pymBk7wySafIxuuYQvTZbYXB+QVCUw0RZs6tzBIUiXPw/wpDWFlkqnn/WgaH1p5W18NLalh3iXZjKoPB7DERCv7sFlGkEuNH3Jq7t0gQnYHBqCAq2B+izsmuv10G+X6Fh5V5tIfU3CqyL4G0iMVVd1eLotcaHFgHa7o+dpIRZ/hXpbV5cSILt1mLGsGVnTaG4PNHyQ==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=46DvMK5GEP4d9GsmJC/9dg2qzjXnaZqUwxzUp3QEP0Q=;
 b=C86ln2gRMKbCx2r/N5IG7p28wurVO8eyCt4CrHdk2fTBuZVPpiL64msCIJ6rZWkH4Bu1XC+UtwdlJltHk4CLIp9nl4uNGxXxJA08osVUCwwn2kYRzoOz+5RtOesUMZRGO0DwDYjigKbK64oGxkanItb9DD5cwwJUPhDubVUCa/cjhNJpvL+R7lnKfT05c6z5PH9JhElKqdk+9BFNkWx2MR7gQpbDiuBUb6X8ddaSCZw7USX/9RA5Y7UJBECPAzFss2vSe9ghGMJSXp0VWXpUJaTwEcxW13l2lLOKdN6/SoreVvaZTs7yrtjs0vuvmeFH74tjlj6AbSXXeNY3CSoTsQ==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=46DvMK5GEP4d9GsmJC/9dg2qzjXnaZqUwxzUp3QEP0Q=;
 b=v8RDlHW/h7TwgQTUiJ0AdF6MbUhKi1H+GcbJ9lwRp2xuS1lYY+O8UU0dEAsgjEZVBamY1kLAKAlQtN1DacfxBAEpdyskAkzN5VHIeGJHA8l5zKtvpLWKSaQVNz85aSIo50AaagVR92YtTbDFjuKX2fpvlI1E1pxXSwYKktz+I7A=
Received: from SN6PR10MB3022.namprd10.prod.outlook.com (2603:10b6:805:d8::25)
 by BL3PR10MB6233.namprd10.prod.outlook.com (2603:10b6:208:38c::18) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.5438.12; Wed, 20 Jul
 2022 02:19:15 +0000
Received: from SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::c4d1:edc3:7d21:7c68]) by SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::c4d1:edc3:7d21:7c68%6]) with mapi id 15.20.5438.024; Wed, 20 Jul 2022
 02:19:15 +0000
From: Liam Howlett <liam.howlett@oracle.com>
To: "maple-tree@lists.infradead.org" <maple-tree@lists.infradead.org>,
        "linux-mm@kvack.org" <linux-mm@kvack.org>,
        "linux-kernel@vger.kernel.org"
	<linux-kernel@vger.kernel.org>,
        Andrew Morton <akpm@linux-foundation.org>,
        Hugh Dickins <hughd@google.com>
CC: Yu Zhao <yuzhao@google.com>
Subject: [PATCH v12 61/69] mm/oom_kill: use maple tree iterators instead of
 vma linked list
Thread-Topic: [PATCH v12 61/69] mm/oom_kill: use maple tree iterators instead
 of vma linked list
Thread-Index: AQHYm97wI+exQWX0Wk+w4FfzqczLYA==
Date: Wed, 20 Jul 2022 02:18:02 +0000
Message-ID: <20220720021727.17018-62-Liam.Howlett@oracle.com>
References: <20220720021727.17018-1-Liam.Howlett@oracle.com>
In-Reply-To: <20220720021727.17018-1-Liam.Howlett@oracle.com>
Accept-Language: en-US
Content-Language: en-US
X-MS-Has-Attach: 
X-MS-TNEF-Correlator: 
x-mailer: git-send-email 2.35.1
x-ms-publictraffictype: Email
x-ms-office365-filtering-correlation-id: 306a1f76-3d60-4ce3-b029-08da69f63e49
x-ms-traffictypediagnostic: BL3PR10MB6233:EE_
x-ms-exchange-senderadcheck: 1
x-ms-exchange-antispam-relay: 0
x-microsoft-antispam: BCL:0;
x-microsoft-antispam-message-info: 
 Vjn/yu9oNKJus+V/ETMVaRPUrW57HLJAFo/bYOm+B6cSCetgcBoy6iVOyzEP8W9bU7uYzON8WUjPrUnsty32Weenqcl5zgM1tWozoAAQ2ldwWO+sBSO4RYYXEy6ozqGAYP7ynbOvzox/lUWu4mTgrmydLC9Gx/LLvStnebD9LAL/3UStOrdqIQz25VDray2u95uMdh2+Uz0pjpUcXf6QiuDerhFsgEZS3r4zS47CfaCZKnLXUc7aiM2qxZCuuSzgHw2U3AIk4569JGU2sjYWn1uQEuffGA/UC+EuKa/Eai7wwPAEkFUpHXKU72SoAUidyLibUpbAzub+sbLCIrAxQRXz4EAqjHh6fu+6TqlGuPPFl+5EnWpYdwx/GZNWfrU5gmgrJiiNkAFFkCVMx7mcvmjvdcy57XWNzm4d7iRqEFnSPdYBhWvYS7XhGrAHpFvheXWOLWDUu8gL50vqF9RnQyxtmchaQV7yQzDQHCxzNYAsaUmjM8+YrKtpE02SLB8oQA9SEsJY1t4JGI6nLzrLE1qN3zyfgbQGcTg3wKx1pwTnj9KFI9R+kfsgbF/7YnUlEwglepcjj/iQyzIKZZ09YS+F/bYmCL1UamKcaOguSEVh523EalFnjqPkETLHnOu46KGCdVBVxiR7+Vq9rhIxR6GlT0HtLhsw/hgVRp7p5Z3Xy7Ei6y5CGdnLWzRiwZm/WjP2GQIXxZD9hBhWAG3KGVmDCjJr181FCfX+1cckwNYC9mNprb+yeEznCkiav0MPFoswoSnoM7mxElfcIng6Hue6mO17szxkQET8DPAh+N5G2EZxjEUWxG6Ctl7aIX+eq+ISwgchOaOSC+gEAdvlfWfB0LfLXwHvqIlx+duQzpWcbF72kEpiRT2RCeXZ757y
x-forefront-antispam-report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3022.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(13230016)(376002)(346002)(396003)(136003)(39860400002)(366004)(186003)(6666004)(2906002)(41300700001)(6512007)(1076003)(38100700002)(6506007)(26005)(2616005)(38070700005)(86362001)(83380400001)(122000001)(478600001)(316002)(44832011)(8676002)(4326008)(64756008)(36756003)(66946007)(5660300002)(8936002)(110136005)(66476007)(91956017)(66446008)(66556008)(76116006)(6486002)(71200400001)(966005);DIR:OUT;SFP:1101;
x-ms-exchange-antispam-messagedata-chunkcount: 1
x-ms-exchange-antispam-messagedata-0: =?iso-8859-1?q?lX9DIFZ4EPKHM3JpDVKKWCR?=
	=?iso-8859-1?q?RTJKRXMPVt5CLLkqP2xxEB1nG4NJS76hL499IxALfT1Qtz4SlY4bd+80uxWz?=
	=?iso-8859-1?q?H7c/HWkVkT/9MjAsYbtpLrlxIImqD016++TnNEf/iDN/ON2JjXn+pGRF8/AM?=
	=?iso-8859-1?q?BH0so1iNOkbhfnlI4dpkvTaqIo2sslBo/JIfkzanm9+wdo2R60ET85S1Imxz?=
	=?iso-8859-1?q?GWylTuus9bhe4kg7/C9x5h+eXW4gzmtUABkc+LrDu56KvRDeQMxrlrvTHRVn?=
	=?iso-8859-1?q?HM9QtFk9BKgwONNQbowXHk78yte5j2gykIcRzIPtNVmNz3hoGzXv1TBF9lOi?=
	=?iso-8859-1?q?Gua8+z0UE7gDW7xTwAsiNRgpQqC6OyWgqSsNpAmkzs7c273ZQVKwR7ssap3A?=
	=?iso-8859-1?q?iA9een+oO9epBNOmbiKQlvL3Us9+sANHiW+OmnclJZ+xvYlxzSurjxQct6M+?=
	=?iso-8859-1?q?E/1HK/GIy1H1rkQyE17F4CMqLN6rpkDUtHkxeuuHqGtvLtN6Zk4Y0yyNb6yI?=
	=?iso-8859-1?q?aD/OnD32/Yt35Q0LQk5HrtGiCuA2MdB2YLnel8FrjOo3VruSOJTBbdGzexWF?=
	=?iso-8859-1?q?2uKFLVfpou6jk2c3w6rBZ5SB8xl040L69w9inU+9AmRUxAuZk/48vFifMglN?=
	=?iso-8859-1?q?bqN13SG9PfPNjmxGQgldMpk/mzydpokAwLXjs34BHLod4nJ476WETExBzXB8?=
	=?iso-8859-1?q?GM+CafMyOq4lZlLg7B7kZION5R09oJqUB9j5MQ/5RGWF8WLoNqUN//d5/8p0?=
	=?iso-8859-1?q?O1ZwT0bCdo65tHPrA5SmB+PW6wM1z1JFcqxaqjm5fUV/XYpkgc5XcVPrwFPB?=
	=?iso-8859-1?q?7SwmlWfhmQhaQuCKNbNKm3rJkvBXKzPf8OE0gKT/E5OPbjv4q8+05iEt+uSr?=
	=?iso-8859-1?q?CUGzpE09Dkee4uEXyHHin6gou3UjvBRCSf0LdcijJfchDAe5vp5LrNOsnQyi?=
	=?iso-8859-1?q?YKwaP5JiIY3gcp5426QwewG8mSjztoXKOoaeJfvcSBmQCYlVNgB+Unzuv+TX?=
	=?iso-8859-1?q?M3QJDmHjX3EHoFgSyL2SxlMidGJCNqMDbrselCXgNOkS7UxujqrT0+yr3s8R?=
	=?iso-8859-1?q?FJtdsGstJrhDcZcR/ka8Crj3jwFFursuqY/jDWb5MGst283AuGF7n5IAPqJZ?=
	=?iso-8859-1?q?G98w8EGSMU/c/Ru+eH8mguhU5nDFR6tkZaf03i4jNT7CTov5dTT6gwZDHSsx?=
	=?iso-8859-1?q?kVd0m5ZxG7LEgDLo9Z6YKXKNFCLkVtaS9FsApyBMe4vtsO1Lr+HmKq3wHslz?=
	=?iso-8859-1?q?wOrziotiwLw7QpeYdX7wEa5tiqLnjxZdsbwM5LHBfkt2PT/J3T2gg2vhWMyC?=
	=?iso-8859-1?q?TiwPM6n/hyl2ftbxQheHbUiuAttHL1jXA1uxVJ+EDhMSpompeOI7WuVznkio?=
	=?iso-8859-1?q?9zwL2rmeA/S29su03cbd3N/p7LPkCKUWSEJglQIGg0lVNZLvZDenpWDSD581?=
	=?iso-8859-1?q?AyQUFsL3/enBQifDjBZ4giIVh3OL7jsZAyXFg12KXpPaaLz/W4VFHNqYZaTh?=
	=?iso-8859-1?q?EeST+b39eUF+Ms/mYeabt45HxTU05N4VHFylIY+1wQtGa9Ua7yQboUY1DhM2?=
	=?iso-8859-1?q?l5hZ/IscxA+QGW5yQk1+KXTmu7yXR4xUjKHDEC0agt8stq8RkJBg2y3sNOXE?=
	=?iso-8859-1?q?g5cPEMUg8/nkdzhuSJiGQRyIu7pMYh0NZnAFipA=3D=3D?=
MIME-Version: 1.0
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3022.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 306a1f76-3d60-4ce3-b029-08da69f63e49
X-MS-Exchange-CrossTenant-originalarrivaltime: 20 Jul 2022 02:18:02.6174
 (UTC)
X-MS-Exchange-CrossTenant-fromentityheader: Hosted
X-MS-Exchange-CrossTenant-id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-mailboxtype: HOSTED
X-MS-Exchange-CrossTenant-userprincipalname: 
 uQKT5DDPGImo9pmE5QoyeepE6gWtMya6mGy8tRg2pX1hgkT/hnO6PIjMoNCEa/nvgiUEKH2DUOgsdSSF8dW8IA==
X-MS-Exchange-Transport-CrossTenantHeadersStamped: BL3PR10MB6233
X-Proofpoint-Virus-Version: vendor=baseguard
 engine=ICAP:2.0.205,Aquarius:18.0.883,Hydra:6.0.517,FMLib:17.11.122.1
 definitions=2022-07-19_10,2022-07-19_01,2022-06-22_01
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 spamscore=0
 bulkscore=0 malwarescore=0
 mlxscore=0 phishscore=0 mlxlogscore=999 suspectscore=0 adultscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2206140000
 definitions=main-2207200008
X-Proofpoint-GUID: 151r8vRkCUfzz-omzrZ8tPb3VMGBwP7-
X-Proofpoint-ORIG-GUID: 151r8vRkCUfzz-omzrZ8tPb3VMGBwP7-
ARC-Seal: i=2; s=arc-20220608; d=hostedemail.com; t=1658283561; a=rsa-sha256;
	cv=pass;
	b=HA/YSGknTxWC05kQezwdWFqXCSBqMx3CrM3wmaAeWFnJYeRBGH5XyECfJ3kjZeJCAGoi+A
	l9dMGMsPvmfqIMQ85wkG8IdEknVLct35x2P4io7t0p1T2SGBP016acwpXJO6y2Hrm0ldyJ
	IcL3pnIQoqNEgZ+Wbf/gCT1QIRiMTIg=
ARC-Authentication-Results: i=2;
	imf16.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=cueXUC4E;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b="v8RDlHW/";
	arc=pass ("microsoft.com:s=arcselector9901:i=1");
	spf=none (imf16.hostedemail.com: domain of liam.howlett@oracle.com has no SPF
 policy when checking 205.220.165.32) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com
ARC-Message-Signature: i=2; a=rsa-sha256; c=relaxed/relaxed;
 d=hostedemail.com;
	s=arc-20220608; t=1658283561;
	h=from:from:sender:reply-to:subject:subject:date:date:
	 message-id:message-id:to:to:cc:cc:mime-version:mime-version:
	 content-type:content-type:
	 content-transfer-encoding:content-transfer-encoding:
	 in-reply-to:in-reply-to:references:references:dkim-signature;
	bh=46DvMK5GEP4d9GsmJC/9dg2qzjXnaZqUwxzUp3QEP0Q=;
	b=oFUBNKpaC6hEgj6OK3B96I9qwwM2M+YORhELO0u33HSn39wGjkIhGOn26JvLSNx8m6XG3n
	G8BXftjnbAJ2k8UZL0Dx+kax7LBm7KF7/hnH7QWsVACnvQewFeDBHditG5XgmV+Vvplkco
	HlsNMJn/5DP4MiUpIAxMIppY8MJyvP4=
X-Stat-Signature: hoorb9ojexr85qx55mhrqcmxb3f791dy
X-Rspamd-Queue-Id: 9B4AF18000A
Authentication-Results: imf16.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=cueXUC4E;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b="v8RDlHW/";
	arc=pass ("microsoft.com:s=arcselector9901:i=1");
	spf=none (imf16.hostedemail.com: domain of liam.howlett@oracle.com has no SPF
 policy when checking 205.220.165.32) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com
X-Rspam-User: 
X-Rspamd-Server: rspam10
X-HE-Tag: 1658283561-454669
X-Bogosity: Ham, tests=bogofilter, spamicity=0.000000, version=1.2.4
Sender: owner-linux-mm@kvack.org
Precedence: bulk
X-Loop: owner-majordomo@kvack.org
List-ID: <linux-mm.kvack.org>

From: "Liam R. Howlett" <Liam.Howlett@Oracle.com>

Link: https://lkml.kernel.org/r/20220504011345.662299-46-Liam.Howlett@oracle.com
Link: https://lkml.kernel.org/r/20220621204632.3370049-62-Liam.Howlett@oracle.com
Signed-off-by: Liam R. Howlett <Liam.Howlett@Oracle.com>
Acked-by: Vlastimil Babka <vbabka@suse.cz>
Cc: Catalin Marinas <catalin.marinas@arm.com>
Cc: David Howells <dhowells@redhat.com>
Cc: "Matthew Wilcox (Oracle)" <willy@infradead.org>
Cc: SeongJae Park <sj@kernel.org>
Cc: Will Deacon <will@kernel.org>
Cc: Davidlohr Bueso <dave@stgolabs.net>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 mm/oom_kill.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mm/oom_kill.c b/mm/oom_kill.c
index 111111111..111111111 100644
--- a/mm/oom_kill.c
+++ b/mm/oom_kill.c
@@ -513,6 +513,7 @@ bool __oom_reap_task_mm(struct mm_struct *mm)
 {
 	struct vm_area_struct *vma;
 	bool ret = true;
+	VMA_ITERATOR(vmi, mm, 0);
 
 	/*
 	 * Tell all users of get_user/copy_from_user etc... that the content
@@ -522,7 +523,7 @@ bool __oom_reap_task_mm(struct mm_struct *mm)
 	 */
 	set_bit(MMF_UNSTABLE, &mm->flags);
 
-	for (vma = mm->mmap ; vma; vma = vma->vm_next) {
+	for_each_vma(vmi, vma) {
 		if (vma->vm_flags & (VM_HUGETLB|VM_PFNMAP))
 			continue;
 

